packages/flutter/test/widgets/modal_barrier_test.dart

```diff
@@ -399,6 +399,22 @@ void main() {
       expect(dismissCallbackCalled, true);
     });
 
+    testWidgets('when onDismiss throws, should have correct context', (tester) async {
+      final barrierKey = UniqueKey();
+      await tester.pumpWidget(MaterialApp(
+        home: Scaffold(
+          body: ModalBarrier(
+            key: barrierKey,
+            onDismiss: () => throw Exception('deliberate'),
+          ),
+        ),
+      ));
+      await tester.tap(find.byKey(barrierKey));
+      await tester.pump();
+      expect(tester.binding.takeException(), 'TODO_assert_the_error_context');
```

Comment:
?

Reply:
Wouldn't the toString of the exception contain something that was added by invokeCallback that you can look for in the test here?

Reply:
Hmm,

```
    print(tester.binding.takeException().toString());
```

yields:

```
Exception: deliberate
```

So I guess it does not add anything.

Reply:
I propose that, in addition to the existing code:

```dart
  dynamic takeException() {
    assert(inTest);
    final dynamic result = _pendingExceptionDetails?.exception;
    _pendingExceptionDetails = null;
    return result;
  }
  FlutterErrorDetails? _pendingExceptionDetails;
```

We can add:

```dart
FlutterErrorDetails takeErrorDetails() {
    assert(inTest);
    final FlutterErrorDetails result = _pendingExceptionDetails; // <-- NOTE
    _pendingExceptionDetails = null;
    return result;
}
```

Reply:
I can create a separate PR for that (given that Flutter guide seems to prefer one PR to do one thing)

Reply:
This shouldn't need new API. Can you get to the error with a custom onError handler? https://master-api.flutter.dev/flutter/foundation/FlutterError/onError.html

Reply:
I will try that. Thanks!